pymeasure/experiment/parameters.py

```
@@ -498,7 +562,137 @@ def __repr__(self):
+    def __str__(self):
+        if not self.is_set():
+            return " "
+        result = {parameter.name: str(parameter) for (_,parameter) in self.parameters.items()}
+        result = json.dumps(result)
+        return result
+        
+class Range1DParameterGroup(ParameterGroup):
+
+    def __init__(self,
+                 name: str,
+                 group_by: Union[GroupByType, None] = {},
```

Ruff (0.15.10)

[warning] Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)